filters/ExpressionFilter.cpp

```diff
@@ -82,48 +82,70 @@ void ExpressionFilter::addArgs(ProgramArgs& args)
 {
     m_args->m_whereArg = &args.add("expression",
         "Conditional expression describing points to be passed to this filter",
-        m_args->m_expression).setPositional();
+        m_args->m_expressions).setPositional();
 }
 
 
 void ExpressionFilter::prepared(PointTableRef table)
 {
-     if (!m_args->m_expression.valid())
-     {
-         std::stringstream oss;
-         oss << "The expression '" <<  m_args->m_expression
-             << "' is invalid";
-         throwError(oss.str());
-     }
-
-     auto status = m_args->m_expression.prepare(table.layout());
+    for (auto& expression: m_args->m_expressions)
+    {
+        if (!expression.valid())
+        {
+            std::stringstream oss;
+            oss << "The expression '" <<  expression
+                << "' is invalid";
+            throwError(oss.str());
+        }
+
+        auto status = expression.prepare(table.layout());
         if (!status)
             throwError("Invalid 'where': " + status.what());
+    }
 }
 
 
 bool ExpressionFilter::processOne(PointRef& point)
 {
-    bool status = m_args->m_expression.eval(point);
+    if (!(m_args->m_expressions.size() == 1))
```

Comment:
I just think the style difference here are funny. (I really laughed.) I would write "!= 1" instead of "(!(==1)". Nothing wrong, but interesting how people think of these things differently.

And below, I'd never use a temporary just to return it.

Wonder why we end up doing the things we do. Some we may have reasons for. Other things, probably not.

Reply:
addressed